src/Plugin/formatter/Formatter.php

```diff
@@ -309,4 +314,23 @@ protected static function unprefixInputOptions($allowed_fields, $prefix) {
     return array_filter(array_map($closure_unprefix, $allowed_fields));
   }
 
+
+  /**
+   * Helper function that calculates the number of items per page.
+   *
+   * @param ResourceInterface $resource
+   *   The associated resource.
+   *
+   * @return int
+   *   The items per page.
+   */
+  protected function calculateItemsPerPage(ResourceInterface $resource = NULL) {
+    $resource = $resource ?: $this->getResource();
```

Comment:
yeah, it's clear you need the resource, my question is why have this `$resource = $resource ?: $this->getResource();` -- instead of forcing passing it as an argument?